sql/analyzer/indexes.go

```diff
@@ -258,6 +260,43 @@ func getIndexes(
 		}
 
 		return result, nil
+	case *expression.Like:
+		// TODO: maybe more cases to simplify
+		r, ok := e.Right.(*expression.Literal)
+		if !ok {
+			break
+		}
+		val := r.Value()
+		valStr, ok := val.(string)
+		if !ok {
+			break
+		}
+		if len(valStr) == 0 {
+			break
+		}
+		if strings.Count(valStr, "_")-strings.Count(valStr, "\\_") > 0 {
+			break
+		}
+		// if there are no wildcards, this is just a plain equals
+		numWild := strings.Count(valStr, "%") - strings.Count(valStr, "\\%")
+		if numWild == 0 {
+			return getIndexes(ctx, ia, expression.NewEquals(e.Left, e.Right), tableAliases)
+		}
+		if len(valStr) >= 2 && valStr[len(valStr)-2:] == "\\%" {
+			break
+		}
+		if valStr[len(valStr)-1] != '%' {
+			break
+		}
+		// TODO: like expression with just a wild card shouldn't even make it here; analyzer rule should just drop filter
+		if len(valStr) == 1 {
```

Comment:
did this error for a test? i'd prefer semantic validation live separately from transforms

Reply:
It didn't error; it's just suboptimal.
`select * from t where x like "%"` is the same as `select * from t`
Currently, we produce a plan that has a `FITER LIKE` node over a table scan

If I didn't have this check here it would just be an index look up over the entire table, which is worse than a table scan (I think).

Reply:
i know you're just trying to get this in for a blog, and i agree with this being better than the index scan over table, adding simplification rules that just do this the right way seems pretty near at hand though. at some point soon we might rewrite the index selection logic, and it's easier if the expression structure has been simplified and standardized upstream. separation of concerns, template for extending expression simplification rules, applies to more queries, ...etc